### ISSUE TITLE
Sometimes uploads can get stuck without feedback to the user

### ISSUE THREAD

**Issue by jornane** [filesender/filesender#187]
I've noticed that some users complain about upload jobs getting stuck. The progress bar just stops and they are left wondering.

From the log file, I get the impression that these users are getting hit by #78 but that they do not get feedback about this for some reason. Sadly this is only a guess as I'm not able to look in the computers of those affected, and I cannot reproduce the bug myself. Bug #35 also comes to mind, because FileSender behaves the same.

Could it be that the client side of FileSender has bad error handling? Maybe we need some kind of [chaos monkey](https://en.wikipedia.org/wiki/Chaos_Monkey) for FileSender developers that will fail a small amount of chunk uploads, so that errors in the client side code are easier to observe.

**Comment by jornane** [comment]
One of our users pointed out that, on Safari, the browser can decide to put background tabs to sleep to save power. He suggested that that could be a cause of uploads getting aborted, because a sleeping tab effectively is the same problems as #35.

**Comment by monkeyiq** [comment]
Thinking out loud, one idea which might not be super for privacy is to be able to optionally enable more logging on these issues. At least knowing the size of the file, what offset the stall happens at, and browser id would be a great start. other metadata could be grabbed at the time server side like average time to transfer bytes so an idea of what sort of pipe the user is connecting over could be guessed at.

**Comment by jornane** [comment]
I don't think we would have big privacy issues logging things, but what do you intend to log? For the server there is no indication that a user closed his laptop, aborted the upload by closing the browser or it simply got stuck. My biggest problem is identifying the issue when it happens.

**Comment by monkeyiq** [comment]
We have some chance here if the upload is not updating the interface and a user wants to help. Perhaps we can put up a button for the user to offer to report it via a "one click".  Mention to the user that clicking the button will allow us to log the transaction number, size of file, where they were in the upload, and browser being used in order to try to replicate these cases and remove them for everybody. Though there is the case that if something in the connection client-server is causing the issue then the logging button itself might not actually connect to the rest server. So the report button might become a mailto: link with subject and body populated. Some way to upload the info out of channel.

**Comment by monkeyiq** [comment]
It seems also Chrome can put background tabs to sleep https://developers.google.com/web/updates/2017/03/background_tabs

**Comment by monkeyiq** [comment]
I took @jornane second comment above and broke that out into another issue https://github.com/filesender/filesender/issues/207 for explicit checking.

**Comment by monkeyiq** [comment]
I recalled this issue after a test I just did, so these two issues are bound:
https://github.com/filesender/filesender/issues/217
